server/src/main/java/org/elasticsearch/inference/ToFilteredXContentObject.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.inference;
+
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Provides a contract writing the full contents of an object as well as a mechanism for
+ * filtering some fields from the response.
+ */
+public interface ToFilteredXContentObject extends ToXContentObject {
```

Comment:
Another pattern you will see in the codebase is to use `params` parameter of `toXContent(XContentBuilder builder, Params params)` to control when the internal fields should be written:

```
if params['include_internal_fields'] {
  // write dimensions_set_by_user
}
```

Both are good, your approach is more explicit which is always good.

Reply:
Ah thanks, good to know 👍